nemo_aligner/utils/train_script_utils.py

```diff
@@ -223,3 +223,56 @@ def temp_pop_from_config(cfg, name_to_temporarily_remove):
             yield
         finally:
             setattr(cfg, name_to_temporarily_remove, temp)
+
+
+def compute_mbs(num_rollout_samples, rollout_micro_batch_size, num_rollout_per_prompt, data_parallel_world_size):
+    '''
+    Batch Size Logic:
+
+    User Specifies:
+    - rollout_micro_batch_size: Number of generations we are able to fit on one worker
+    - num_rollout_samples: This is the number of generations we want to generate in each pass
+    - num_rollout_per_prompt: The number of rollout samples for any given prompt
+
+    We need to get:
+    - mbs: The number of prompts to load per worker
+    - duplicate: The number of times to duplicate each prompt
+    - generation_iter: number of iterations to run generation for
+    
+    if num_rollout_per_prompt * 2 < rollout_micro_batch_size:
+        duplicate = num_rollout_per_prompt
+        mbs = rollout_micro_batch_size / num_rollout_per_prompt # We want to fill out the microbatch size with as many prompts as possible
+        generation_iter = 1
+    else:
+        mbs = 2 # In this case we want to generate more responses than our micro_batch_size, so we have to split the generation up across devices/iterations. However, 2 is the minimum size allowed
+        duplicate = rollout_micro_batch_size / 2
+        generation_iter =  num_rollout_per_prompt / duplicate
+        
+    
+    Constraints:
+    C1: gbs % mbs = 0
+    C2: Either rollout_micro_batch_size % N == 0 or N % rollout_micro_batch_size == 0
+    C3: cfg.model.ppo.num_rollout_per_prompt % DP == 0
+
+    '''
+
+    if num_rollout_samples % rollout_micro_batch_size != 0:
```

Comment:
for this function generally -> can you use https://github.com/NVIDIA/Megatron-LM/blob/72008a0460000360ae4542b5411f25175d899b2e/megatron/core/utils.py#L34?

Reply:
Changed this function to use divide.

Reply:
I believe this function is still needed, as there are two cases to consider when calculating the mbs.

Reply:
hmm, sorry maybe i misunderstand -- why not just stack everything like you're doing in RStrainer and then try our best to cut the batch down to rollout mbs? if it's not divisible cleanly we can do a min

Reply:
I removed compute_mbs and stack everything in RStrainer like mentioned.